packages/components/src/DateTimePickers/pickers/DatePicker/DatePicker.test.js

```diff
@@ -1,14 +1,572 @@
+
+		const wrapper = shallow(<DatePicker
+			calendar={calendar}
+			selectedDate={selectedDate}
+			onSelect={() => {}}
+			disabledRules={disabledDates}
+		/>);
 
-		// then
 		expect(wrapper.getElement()).toMatchSnapshot();
 	});
+
+	describe('body calendar', () => {
+		describe('first date in the correct day column (Monday as first column)', () => {
+			function getFirstRowCellText(wrapper, column) {
+				return wrapper
+					.find('.theme-calendar-body-row')
+					.first()
+					.find('.theme-calendar-item')
+					.at(column - 1)
+					.find('DayPickerAction')
+					.prop('label');
+			}
+
+			it('should have monday in the 1st column for january 2018', () => {
+				const calendar = {
+					year: 2018,
+					monthIndex: 0,
+				};
+
+				const wrapper = shallow(<DatePicker
+					calendar={calendar}
+					onSelect={() => {}}
+				/>);
+
+				expect(getFirstRowCellText(wrapper, 1)).toBe('1');
+			});
+
+			it('should have thursday in the 4th column for march 2018 ', () => {
+				const calendar = {
+					year: 2018,
+					monthIndex: 2,
+				};
+
+				const wrapper = shallow(<DatePicker
+					calendar={calendar}
+					onSelect={() => {}}
+				/>);
+
+				expect(getFirstRowCellText(wrapper, 4)).toBe('1');
+			});
+
+			it('should have sunday in the 7th column for april 2018', () => {
+				const calendar = {
+					year: 2018,
+					monthIndex: 3,
+				};
+
+				const wrapper = shallow(<DatePicker
+					calendar={calendar}
+					onSelect={() => {}}
+				/>);
+
+				expect(getFirstRowCellText(wrapper, 7)).toBe('1');
+			});
+		});
+
+
+		describe('last date in the correct day column (Monday as first column)', () => {
+			function getLastRowCellText(wrapper, column) {
+				return wrapper
+					.find('.theme-calendar-body-row')
+					.last()
+					.find('.theme-calendar-item')
+					.at(column - 1)
+					.find('DayPickerAction')
+					.prop('label');
+			}
+
+			it('should have monday in the 1st column for april 2018', () => {
+				const calendar = {
+					year: 2018,
+					monthIndex: 3,
+				};
+
+				const wrapper = shallow(<DatePicker
+					calendar={calendar}
+					onSelect={() => {}}
+				/>);
+
+				expect(getLastRowCellText(wrapper, 1)).toBe('30');
+			});
+
+			it('should have thursday in the 4th column for may 2018', () => {
+				const calendar = {
+					year: 2018,
+					monthIndex: 4,
+				};
+
+				const wrapper = shallow(<DatePicker
+					calendar={calendar}
+					onSelect={() => {}}
+				/>);
+
+				expect(getLastRowCellText(wrapper, 4)).toBe('31');
+			});
+
+			it('should have sunday in the 7th column for september 2018', () => {
+				const calendar = {
+					year: 2018,
+					monthIndex: 8,
+				};
+
+				const wrapper = shallow(<DatePicker
+					calendar={calendar}
+					onSelect={() => {}}
+				/>);
+
+				expect(getLastRowCellText(wrapper, 7)).toBe('30');
+			});
+		});
+
+		describe('number of weeks displayed', () => {
+			function nbOfWeeksRendered(wrapper) {
+				return wrapper.find('.theme-calendar-body-row').length;
```

Comment:
Never rely on `theme-my-classname` classNames. Those are generated with some config.
Add the `my-classname` in addition to the css module in you code. So your test depends in your code, and not the test configuration

Reply:
✔

Reply:
We have a sort of convention to have the classnames be like
`tc-${component-name}-${classname-qualifiers}`

tc: Talend-Component
(If forms package we have tf)

Reply:
✔